plasmapy/plasma/grids.py

```diff
@@ -1105,16 +1108,16 @@ def volume_averaged_interpolator(
 
         # Get the indices that are equal to nan (fill values), then set
         # their values to 0. They will be over-written after the interpolation
-
-        # Nan array is shape [n] and is 1 if none of the indices for a
-        # position are NaN, and 0 otherwise.
-        nan_mask = np.where(np.isnan(np.sum(i, axis=1)), 0, 1)
+        nan_mask = np.isnan(i).any(axis=1)
         # Replace all NaNs temporarily with 0
         i = np.where(np.isnan(i), 0, i)
         i = i.astype(np.int32)  # Cast as integers
 
         # Load grid attributes (so this isn't repeated)
         ax0, ax1, ax2 = self.ax0.si.value, self.ax1.si.value, self.ax2.si.value
+        dx, dy, dz = self.dax0.si.value, self.dax1.si.value, self.dax2.si.value
```

Comment:
Unless you're trying to enforce/check some kind of units here, I'd stick with using `self.ds.coords["ax0"].values` since the adding and stripping of units incurs a non-negligible time penalty.

Reply:
Unfortunately the units of the axes aren't known and we need to make sure they match the provided position units, so I think this is necessary. 

I could use .to(u.m), given that this is Cartesian coordinates: I think that's faster.

Reply:
Ok. If I remember correctly, that method is faster.

It might be faster to do something more like...

```python

if isinstance(pos, u.Qantity)
    pos_unit = pos.unit
    pos = pos.value
else:
    pos_unit = self.unit

....

if pos_unit == self.unit:
    # assuming you're enforcing units[0] == units[1] == units[2]
    ax0 = self.ds.coords["ax0"]
    ...
else:
    ax0 = self.ax0.to(u.m).value
    ...
```

It might even be worthwhile to have a unitless version of the "ax0" properties...

```python
@property
def ax0_unitless(self):
    if self.is_uniform:
        return self.ds.coords["ax0"].values
    else:
        raise ValueError(...)

@property
def ax0(self):
    return self.ax0_unitless * self.unit0
```

I only suggest this so you don't have to keep adding and stripping unis for calculations, and the functionality is wrapped into an interface so it will be easier to make modifications (if needed) without breaking a bunch of code.

Reply:
I was thinking about making private methods without units: I will try that. 

What do you think about making many of these axes cached properties? They always return the same thing unless someone manually messes with the coordinates in `self.ds` which is not a supported way of interacting with the object.

Reply:
... I'm reluctant to that since they are public attributes and thus we just create the potential for miss use and unintended behavior.  If they were private methods, then I wouldn't be as concerned.

Reply:
I did run into an issue w/ the implementation of the first code block because `self.interpolate_indices(pos)` is called on line 1134 and it's needing units to return the correct value.

Reply:
Yeah, this is a little complicated because we need to reconcile units on the grid and in user-provided positions. 

I think this is non-trivial and should be a separate PR in the future if we want to get into it. I'm also not as concerned about the speed enhancements here, because the vast majority of time is spent in the scipy interpolator.